tests/acceptance/test_sro_measures_dsl.py

```diff
@@ -0,0 +1,157 @@
+from databuilder import Measure
+from databuilder.tables import clinical_events, practice_registrations
+
+from .codelists import (
+    alt_codelist,
+    asthma_codelist,
+    cholesterol_codelist,
+    copd_codelist,
+    hba1c_codelist,
+    medication_review_codelist,
+    qrisk_codelist,
+    rbc_codelist,
+    sodium_codelist,
+    systolic_bp_codelist,
+    tsh_codelist,
+)
+
+
+def test_sro_measures_study(cohort, subtest):
+    index_date = "2020-11-01"
+    end_of_index_month = "2020-11-30"
+
+    measures_codelists = {
+        "medication_review": medication_review_codelist,
+        "systolic_bp": systolic_bp_codelist,
+        "qrisk": qrisk_codelist,
+        "cholesterol": cholesterol_codelist,
+        "alt": alt_codelist,
+        "tsh": tsh_codelist,
+        "rbc": rbc_codelist,
+        "hba1c": hba1c_codelist,
+        "sodium": sodium_codelist,
+        "asthma": asthma_codelist,
+        "copd": copd_codelist,
+    }
+
+    with subtest("population"):
+        filtered_practice_registrations = (
+            practice_registrations.filter(
+                practice_registrations.date_start <= index_date
+            )
+            # TODO: Reinstate the following when Predicates support logic operations
+            # .filter(
+            #     (practice_registrations.date_end >= index_date)
+            #     | (practice_registrations.date_end != None)
+            # )
+            .filter(practice_registrations.date_end >= index_date)
+        )
+
+        is_registered = filtered_practice_registrations.exists_for_patient()
+
+        # TODO: The following would be nice
+        # not_dead = patient_demographics.date_of_death == None
+
+        # TODO: Does this work as we expect it to work?
+        # not_dead = (
+        #     patient_demographics.select_column(patient_demographics.date_of_death)
+        #     == None  # noqa: E711
+        # )
+
+        # TODO: This does not work, because population.value is an instance of
+        # Comparator
+        # cohort.set_population(is_registered & not_dead)
```

Comment:
There's something weird going on here. It's a bug, but I'm not sure that it's worth fixing since this will be overhauled soon. @inglesp I think this is probably related to the thing I ran into yesterday.

Reply:
I tried several times to understand what was happening, but kept arriving at the definition of `Comparator` and not really understanding what was being compared to what. How is `Comparator` being overhauled?

Reply:
`Comparator` is part of the old query model. That's going to be replaced.